ansible/roles/nginx/files/genssl.sh

```diff
@@ -8,22 +8,71 @@ if [ "$#" -ne 1 ]; then
     echo "usage: $0 [common name: host or ip]"
 fi
 CN=$1
+TYPE=$2
+PASSWORD="openwhisk"
 
 ## generates a (self-signed) certificate
 
 ## uncomment to regenerate the key
-#openssl genrsa -out "$SCRIPTDIR/openwhisk-key.pem" 2048
+#openssl genrsa -out "$SCRIPTDIR/openwhisk-server-key.pem" 2048
 
-echo generating certificate request
-openssl req -new \
-    -key "$SCRIPTDIR/openwhisk-key.pem" \
-    -nodes \
+if [ "$TYPE" == "server" ]; then
+    echo generating server certificate request
+    openssl req -new \
+        -key "$SCRIPTDIR/openwhisk-server-key.pem" \
```

Comment:
If custom `{{ nginx.ssl.cert }}` is configured, certificate generation step will be skipped.